Packs/OPSWAT-Filescan/Integrations/OPSWATFilescan/OPSWATFilescan.py

```diff
@@ -0,0 +1,390 @@
+urllib3.disable_warnings()
+
+"""GLOBALS/PARAMS """
+INTEGRATION_NAME = "OPSWAT Filescan Integration"
+INTEGRATION_CONTEXT_NAME = "OPSWAT.Filescan"
+
+
+class Client(BaseClient):
+    def __init__(self, api_key: str, base_url: str, proxy: bool, verify: bool):
+        super().__init__(base_url=base_url, proxy=proxy, verify=verify)
+        self.api_key = api_key
+
+        if self.api_key:
+            self._headers = {"X-Api-Key": self.api_key}
+
+    def test_module(self) -> Dict:
+        """
+        Return information about the user. (Need API key)
+        """
+        request_result = self._http_request(
+            method="GET",
+            ok_codes=([200]),
+            url_suffix="/users/me",
+        )
+        return request_result
+
+    def post_sample(self, args: Dict[str, Any]) -> Dict[str, Any]:
+        data = {}
+
+        if description := args.get("description"):
+            data["description"] = description
+        if tags := args.get("tags"):
+            data["tags"] = tags
+        if password := args.get("password"):
+            data["password"] = password
+        if is_private := args.get("is_private"):
+            data["is_private"] = is_private
+
+        if url := args.get("url"):
+            data["url"] = url
+
+            return self._http_request(
+                method="POST", url_suffix="/scan/url", ok_codes=([200]), data=data
+            )
+
+        elif entry_id := args.get("entry_id"):
+            try:
+                file_entry = demisto.getFilePath(entry_id)
+            except Exception as e:
+                raise DemistoException(
+                    f'Failed to find file entry with id:"{entry_id}". got error: {e}'
+                )
+
+            with open(file_entry["path"], "rb") as file:
+                return self._http_request(
+                    method="POST",
+                    url_suffix="/scan/file",
+                    ok_codes=([200]),
+                    data=data,
+                    files={"file": file},
+                )
+        else:
+            raise DemistoException("No file or URL was provided.")
+
+    def get_scan_result(self, flow_id: str) -> Dict[str, Any]:
+
+        filters = [
+            "filter=general",
+            "filter=finalVerdict",
+            "filter=allTags",
+            "filter=overallState",
+            "filter=taskReference",
+            "filter=subtaskReferences",
+            "filter=allSignalGroups",
+        ]
+
+        postfix = "&".join(filters)
+
+        url_suffix = f"/scan/{flow_id}/report?{postfix}"
+
+        response = self._http_request(
+            method="GET",
+            ok_codes=([200]),
+            url_suffix=url_suffix,
+        )
+
+        return response
+
+    def get_search_query(self, query_str: str) -> Dict[str, Any]:
+        return self._http_request(
+            method="GET",
+            ok_codes=([200]),
+            params={"query": query_str, "page_size": 10, "page": 1},
```

Comment:
1) what's the reason we always requesting for page 1? 
2) what's the reason we always requesting for page_size 10?

Reply:
In this first version of the integration, with the search request, we always give back the first 10 responses from the 1st page.    (This may change in future versions.)